dragonfly/roompolygon.py

```diff
@@ -0,0 +1,421 @@
+# coding: utf-8
+"""Dragonfly Room Polygon."""
+from ._base import _Base
+from .properties import RoomPolygonProperties
+from .glazingparameter import GlazingParameter
+
+import honeybee.boundarycondition.boundary_conditions as bcs
+from honeybee.boundarycondition import _BoundaryCondition, Outdoors, Surface
+from honeybee.typing import tuple_with_length, float_positive
+from honeybee.room import Room
+from honeybee.aperture import Aperture
+
+from ladybug_geometry.geometry3d.face import Face3D
+from ladybug_geometry.geometry3d.polyface import Polyface3D
+
+import math
+
+
+class RoomPolygon(_Base):
+    """A volume defined by an extruded floor plate, representing a single room or space.
+
+    Properties:
+        * name
+        * display_name
+        * geometry
+        * floor_to_ceiling_height
+        * boundary_conditions
+        * glazing_ratios
+        * glazing_parameters
+        * parent
+        * has_parent
+        * floor_segments
+        * segment_count
+        * floor_height
+        * ceiling_height
+        * volume
+        * floor_area
+        * exterior_wall_area
+        * exterior_aperture_area
+    """
+    __slots__ = ('_geometry', '_segment_count', '_floor_to_ceiling_height',
+                 '_boundary_conditions', '_glazing_ratios', '_glazing_parameters',
+                 '_parent')
+
+    def __init__(self, name, geometry, floor_to_ceiling_height, boundary_conditions=None,
+                 glazing_ratios=None, glazing_parameters=None, tolerance=None):
+        """A volume defined by an extruded floor plate, representing a single room.
+
+        Args:
+            name: Room name. Must be < 100 characters.
+            geometry: A single Face3D object representing the floor plate of the Room.
+            floor_to_ceiling_height: A number for the height above the floor where the
+                ceiling begins. Typical values range from 3 to 5 meters.
+            boundary_conditions: A list of boundary conditions that match the number of
+                segments in the input Face3D geometry. These will be used to assign
+                boundary conditions to each of the walls of the Room in the resulting
+                model. If None, all boundary conditions will be Outdoors or Ground
+                depending on whether ceiling of the room is below 0 (the assumed
+                ground plane).
+            glazing_ratios: A list of numbers between 0 and 1 that match the number of
```

Comment:
Now that I see this I get the answer to the other comment that I left on glazing parameters class. Is there a reason not to include glazing ratio as part of the glazing parameters object?

Reply:
There are many reasons why it's preferable to separate the amount of glass from the instructions for how to shape it. The first is ease in calculating metrics like total glazing area over a building. The second is not having to store information about how to shape the glass if you do not care about it (which is probably the case for most energy models). I was just taking a look at how floorspace.js does it, though, and I realize that they supported a few different ways to define window geometries (closer to what you have been thinking here).  However, their methods are definitely prone to errors if the user isn't checking what they put in. Maybe we should draw up a list of pros/cons and discuss.

Reply:
I am going to resolve this sine I plan to implement it as you suggest here.